cmake/CheckLibc.cmake

```diff
@@ -29,3 +29,23 @@ if (Strerror_R_Returns_Char_P)
     PRIVATE
       SEASTAR_STRERROR_R_CHAR_P)
 endif ()
+
+check_cxx_source_compiles ("
+#define _GNU_SOURCE
+#include <pthread.h>
+#include <sched.h>
+
+int main() {
+    pthread_attr_t pa;
+    pthread_attr_init(&pa);
+    cpu_set_t cpuset;
+    pthread_attr_setaffinity_np(&pa, sizeof(cpuset), &cpuset);
+}"
+  Pthread_Attr_Setaffinity_Np)
+if (Pthread_Attr_Setaffinity_Np)
+  # define Pthread_Attr_Setaffinity_Np  if provided as is GNU-specific version,
```

Comment:
what is defined is `SEASTAR_PTHREAD_ATTR_SETAFFINITY_NP`, so might want to update the comment accordingly. but probably we can drop the comment. as the difference is less subtle than that between the two variants of `strerror_r()` -- they have different function signatures and usage.

Reply:
dropped the comment